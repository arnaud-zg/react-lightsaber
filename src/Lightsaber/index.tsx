import React from 'react'
import './style.scss'

export enum ELightsaberOwners {
  YODA = 'yoda',
  VADER = 'vader',
  WINDU = 'windu',
  OBI = 'obi-wan',
}

interface IProps {
  id: string
  owner: ELightsaberOwners
}

const Lightsaber = (props: IProps) => {
  const { id, owner } = props

  return (
    <div className="lightsaber-component">
      <input type="checkbox" id={id} className={`jedi jedi--${owner}`} />
    </div>
  )
}

export default Lightsaber
