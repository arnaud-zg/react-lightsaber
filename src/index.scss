@import './_common/variable.scss';
@import './Lightsaber/index.scss';
