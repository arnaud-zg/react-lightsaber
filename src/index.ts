export { default as Lightsaber } from './Lightsaber/index'

export const life = (): { status: string } => ({ status: 'ok' })

export default {}
