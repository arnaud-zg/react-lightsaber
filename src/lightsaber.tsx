import React from 'react'

const Lightsaber = () => <div>Lightsaber</div>

export default Lightsaber
