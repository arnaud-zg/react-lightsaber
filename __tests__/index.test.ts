import { life } from '../src/index'

describe('Method.life()', () => {
  it('should return a valid object', () => {
    expect(life()).toBeTruthy()
    expect(life()).toMatchSnapshot()
  })
})
