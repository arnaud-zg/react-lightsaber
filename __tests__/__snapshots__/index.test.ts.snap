// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Method.life() should return a valid object 1`] = `
Object {
  "status": "ok",
}
`;
